src/test/java/io/vertx/ext/cluster/infinispan/test/InfinispanClusterWideMapTest.java

```diff
@@ -57,4 +69,127 @@ protected void clusteredVertx(VertxOptions options, Handler<AsyncResult<Vertx>>
   protected ClusterManager getClusterManager() {
     return new InfinispanClusterManager();
   }
+
+  @Test
+  @Repeat(times = 10)
+  public void testKeyStream() {
+    Map<JsonObject, Buffer> map = genJsonToBuffer(100);
+    loadData(map, (vertx, asyncMap) -> {
+      List<JsonObject> keys = new ArrayList<>();
+      ReadStream<JsonObject> stream = InfinispanAsyncMap.<JsonObject, Buffer>unwrap(asyncMap).keyStream();
+      long pause = 500;
+      Long start = System.nanoTime();
+      stream.endHandler(end -> {
+        assertEquals(map.size(), keys.size());
+        assertTrue(keys.containsAll(map.keySet()));
+        long duration = NANOSECONDS.toMillis(System.nanoTime() - start);
+        assertTrue(duration >= 3 * pause);
+        testComplete();
+      }).exceptionHandler(t -> {
+        fail(t);
+      }).handler(jsonObject -> {
+        keys.add(jsonObject);
+        if (jsonObject.getInteger("key") == 3 || jsonObject.getInteger("key") == 16 || jsonObject.getInteger("key") == 38) {
+          stream.pause();
+          int emitted = keys.size();
+          vertx.setTimer(pause, tid -> {
+            assertTrue("Items emitted during pause", emitted == keys.size());
+            stream.resume();
+          });
+        }
+      });
+    });
+    await();
+  }
+
+  @Test
+  @Repeat(times = 10)
+  public void testValueStream() {
+    Map<JsonObject, Buffer> map = genJsonToBuffer(100);
+    loadData(map, (vertx, asyncMap) -> {
+      List<Buffer> values = new ArrayList<>();
+      ReadStream<Buffer> stream = InfinispanAsyncMap.<JsonObject, Buffer>unwrap(asyncMap).valueStream();
+      AtomicInteger idx = new AtomicInteger();
+      long pause = 500;
+      Long start = System.nanoTime();
+      stream.endHandler(end -> {
+        assertEquals(map.size(), values.size());
+        assertTrue(values.containsAll(map.values()));
+        assertTrue(map.values().containsAll(values));
+        long duration = NANOSECONDS.toMillis(System.nanoTime() - start);
+        assertTrue(duration >= 3 * pause);
+        testComplete();
+      }).exceptionHandler(t -> {
+        fail(t);
+      }).handler(buffer -> {
+        values.add(buffer);
+        int j = idx.getAndIncrement();
+        if (j == 3 || j == 16 || j == 38) {
+          stream.pause();
+          int emitted = values.size();
+          vertx.setTimer(pause, tid -> {
+            assertTrue("Items emitted during pause", emitted == values.size());
+            stream.resume();
+          });
+        }
+      });
+    });
+    await();
+  }
+
+  @Test
+  @Repeat(times = 10)
+  public void testEntryStream() {
```

Comment:
three tests look very similar, perhaps you can make one generic test and use a variable to customize it ?

Reply:
Done